ChopChop/ViewModels/Users/ProfileViewModel.swift

```diff
@@ -0,0 +1,84 @@
+import SwiftUI
+import Combine
+
+final class ProfileViewModel: ObservableObject {
+    private let storageManager = StorageManager()
+    private let userId: String
+    private let settings: UserSettings
+    @ObservedObject private(set) var recipesViewModel: OnlineRecipeCollectionViewModel
+
+    private var ownUserCancellable: AnyCancellable?
+    private var userCancellable: AnyCancellable?
+    private var recipesCancellable: AnyCancellable?
+
+    @Published private(set) var userName = ""
+    @Published private(set) var publishedRecipesCount = 0
+    @Published private(set) var followeeCount = 0
+
+    private var followeeIds: [String] = []
+    @Published private(set) var isFollowedByUser = false
+
+    init(userId: String, settings: UserSettings) {
+        self.userId = userId
+        self.settings = settings
+        self.recipesViewModel = OnlineRecipeCollectionViewModel(
+            publisher: ProfileViewModel.getRecipesPublisher(userId: userId))
+
+        ownUserCancellable = ownUserPublisher?
+            .sink { [weak self] ownUser in
+                self?.followeeIds = ownUser.followees
+                self?.isFollowedByUser = ownUser.followees.contains(userId)
+            }
+
+        userCancellable = userPublisher
+            .sink { [weak self ] user in
+                self?.userName = user.name
+                self?.followeeCount = user.followees.count
+            }
+
+        recipesCancellable = recipesViewModel.$recipes
+            .sink { [weak self] recipes in
+                self?.publishedRecipesCount = recipes.count
+            }
+    }
+
+    var isOwnProfile: Bool {
+        userId == settings.userId
+    }
+
+    func addFollowee() {
+        guard let ownId = settings.userId, !isOwnProfile, !isFollowedByUser else {
+            return
+        }
+
+        storageManager.addFollowee(userId: ownId, followeeId: userId)
+    }
+
+    func removeFollowee() {
+        guard let ownId = settings.userId, !isOwnProfile, isFollowedByUser else {
+            return
+        }
+
+        storageManager.removeFollowee(userId: ownId, followeeId: userId)
+    }
+
+    private var ownUserPublisher: AnyPublisher<User, Never>? {
+        guard let ownId = settings.userId else {
+            return nil
+        }
+
+        return storageManager.userByIdPublisher(userId: ownId)
+            .assertNoFailure()
+            .eraseToAnyPublisher()
+    }
+
+    private var userPublisher: AnyPublisher<User, Never> {
+        storageManager.userByIdPublisher(userId: userId)
+            .assertNoFailure()
+            .eraseToAnyPublisher()
+    }
+
+    private static func getRecipesPublisher(userId: String) -> AnyPublisher<[OnlineRecipe], Error> {
+        StorageManager().allRecipesByUsersPublisher(userIds: [userId])
```

Comment:
maybe can set StorageManager as a private variable

Reply:
Issue with that is that this publisher is required before every property in the view model is initialised, so using `self.storageManager` is not allowed